pkg/integrations/oracledb_exporter/oracledb_exporter.go

```diff
@@ -0,0 +1,114 @@
+package oracledbexporter
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/pkg/integrations"
+	oe "github.com/observiq/oracledb_exporter/collector"
+
+	// required driver for integration
+	_ "github.com/sijms/go-ora/v2"
+	"github.com/xo/dburl"
+)
+
+// DefaultConfig is the default config for the oracledb v2 integration
+var DefaultConfig = Config{
+	ConnectionString:     os.Getenv("DATA_SOURCE_NAME"),
+	MaxOpenConns:         10,
+	MaxIdleConns:         0,
+	QueryTimeout:         5,
+	MetricScrapeInterval: 0,
+}
+
+var (
+	errNoConnectionString = errors.New("no connection string was provided")
+	errNoHostname         = errors.New("no hostname in connection string")
+)
+
+// Config is the configuration for the oracledb v2 integration
+type Config struct {
+	ConnectionString     string        `yaml:"connection_string"`
+	MaxIdleConns         int           `yaml:"max_idle_connections"`
+	MaxOpenConns         int           `yaml:"max_open_connections"`
+	MetricScrapeInterval time.Duration `yaml:"metric_scrape_interval"`
+	QueryTimeout         int           `yaml:"query_timeout"`
+}
+
+// Validate returns if the configuration is valid
+func (c *Config) Validate() error {
+	if c.ConnectionString == "" {
+		return errNoConnectionString
+	}
+	u, err := dburl.Parse(c.ConnectionString)
```

Comment:
Is there any reason to not to use standard `url.Parse`?

Reply:
Yeah it might be a sledgehammer usage where a hammer works kind of situation. I ended up removing it and using the standard `net/url` package.